text/0058-incoming-request-qos.md

```diff
@@ -0,0 +1,63 @@
+# Incoming request quality-of-service
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0058](https://github.com/casperlabs/ceps/pull/0058)
+
+To ensure stable and fair service of requests even under load, we throttle incoming traffic from a peer to a fixed rate of requests. Peers are subjected to similar restrictions as those for outbound bandwidth described in [CEP42](0042-network-qos.md), although instead of bytes we use a request count.
+
+## Motivation
+
+[motivation]: #motivation
+
+By restricting the number of requests, we ensure a high quality of service (i.e. low block times), even when a lot of new users are joining the system. By reserving resources for active validators, this approach scales to a large number of newly joining nodes in case of a spike of interest in the network.
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+Each incoming request is subtracted from a periodically refreshed per-peer allowance (see [CEP42](0042-network-qos.md) for details). Should a node exceed this allowance, incoming messages are temporarily delayed and no further data is read from the peers connection, causing regular backpressure through the underlying TCP/IP connection. It is important to note that in this case the burden of buffering requests is on the requesting peer, which is ideal.
+
+Initially, only `GetRequest` messages, which typically make up the bulk of the traffic when joining, are subject to this limitation, although this can easily expanded to other network message types.
```

Comment:
```suggestion
Initially, only `GetRequest` messages, which typically make up the bulk of the traffic when joining, are subject to this limitation, although this can easily be expanded to other network message types.
```